### ISSUE TITLE
Some Automation Tasks for Safari break terminate my workflow  if no Safari Window open

### ISSUE THREAD

**Issue by Hanspe-3019** [alfredapp/automation-tasks#10]
"Current Safari Tab", "Safari Tab List" or "Close Current Safari Tab" break when used against Safari with no window open. I suggest a reasonable output in these cases like empty list or empty string, so I do not have to make extra test before using such tasks.

To reproduce:

- Build simple worklflow with Automation Task Current Safari Tab" piped into Debug.

- Run with several Safari Windows open:
```
[16:42:41.309] test[Keyword] Processing complete
[16:42:41.314] test[Keyword] Passing output '' to Automation Task
[16:42:41.315] test[Automation Task] Running task 'Current Safari Tab' with no arguments
[16:42:41.863] test[Automation Task] Processing complete
[16:42:41.867] test[Automation Task] Passing output 'https://github.com/alfredapp/automation-tasks/issues/9' to Debug
```

- Now Close all Windows and run again:
```
[16:42:41.868] test[Debug] 'https://github.com/alfredapp/automation-tasks/issues/9', {}
[16:43:41.341] test[Keyword] Processing complete
[16:43:41.342] test[Keyword] Passing output '' to Automation Task
[16:43:41.344] test[Automation Task] Running task 'Current Safari Tab' with no arguments
[16:43:41.520] ERROR: test[Automation Task] Task Error: No windows found.
```
Task Error terminates my workflow!
Compare with this the next test case:

- Quit Safari and run again:
```
[16:44:07.126] test[Keyword] Processing complete
[16:44:07.128] test[Keyword] Passing output '' to Automation Task
[16:44:07.129] test[Automation Task] Running task 'Current Safari Tab' with no arguments
[16:44:09.999] test[Automation Task] Processing complete
[16:44:10.004] test[Automation Task] Passing output '<null>' to Debug
[16:44:10.006] test[Debug] '<null>', {}
```

Here the task returns a reasonable output, that is ok.

---

Alfred 5.1.4 and Automation Tasks 2023.8
Safari 17.2 and macOS 13.6.3 (22G436)

**Comment by vitorgalvao** [comment]
I’m rethinking the approach here. While in some cases it makes sense to continue even if nothing is found (closing current tab), in others (getting current tab) it does make sense to stop, because otherwise the workflow may not do what is expected and those people will need to do their own check afterwards.

Could you expand on your real use case, so I have more data to think with?

**Comment by Hanspe-3019** [comment]
My real use case is a little helper to switch to a new Private/Public/Profile window:

The workflow starts with a List Filter where I choose the desired window type.

If Safari is the front app, it passes the URL of the frontmost tab, close this tab and pipe the URL to open it in a new created window of the desired type. Otherwise the workflow finally opens an empty window.

The automation task *Close Current Safari Tab* would do the job, provided it would tolerate the stupid case that no window exists instead of terminating the workflow. So I had to implement this functionality myself with a run script:

```applescript
on run argv
  tell application "Safari"
	if (count of windows) > 0 then
		set ct to current tab of window 1
		set theURL to URL of ct
		tell ct to close
		return theURL
	end if
  end tell
  return "*no window*"
end run 
```
I think the automation tasks should only throw an exception if a situation has arisen that cannot be handled by the workflow in a meaningful way. It is sufficient if the documentation for a task describes the output in such edge cases, example "Output: URL of closed tab or empty string if no tab exists."

**Comment by vitorgalvao** [comment]
Both cases are valid and should be supported. Just released 2023.9.

In the Browser Automation Tasks you can now find an `Allow empty result when no windows are open`. It also does more checks (e.g. if the browser is not running) if that box is unticked. For your use case you want to turn the checkbox on, but I’ve made it off by default to also better support workflows made before this update.
